src/statmodels.jl

```diff
@@ -160,17 +160,50 @@ df_residual(obj::RegressionModel) = error("df_residual is not defined for $(type
 ## Nms are the coefficient names, corresponding to rows in the table
 type CoefTable
     mat::Matrix
+    cols::Tuple
     colnms::Vector
     rownms::Vector
-    pvalcol::Integer
-    function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
-        nr,nc = size(mat)
-        0 <= pvalcol <= nc || error("pvalcol = $pvalcol should be in 0,...,$nc]")
+    function CoefTable(cols::Tuple,colnms::Vector,rownms::Vector)
+        nc = length(cols)
+        nrs = map(length,cols)
+        nr = max(nrs...)
         length(colnms) in [0,nc] || error("colnms should have length 0 or $nc")
         length(rownms) in [0,nr] || error("rownms should have length 0 or $nr")
-        new(mat,colnms,rownms,pvalcol)
+        if all(unique(nrs) .== nr)
+            mat = hcat(cols...)
+        else
+            mat = Matrix(nr,nc) # A better value here?
+        end
+        new(mat,cols,colnms,rownms)
+    end
+end
+function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
+    nr,nc = size(mat)
+    cols = Any[mat[:, i] for i in 1:nc]
+    if pvalcol != 0                         # format the p-values column
+        cols[pvalcol] = PValue(cols[pvalcol])
+    end
+    cols = ntuple((i)->cols[i],nc)
+    CoefTable(cols,colnms,rownms)
+end
+
+type PValue
+    v::Number
+    function PValue(v::Number)
+        0. <= v <= 1. || error("p-values must be in [0.,1.]")
+        new(v)
+    end
+end
+function PValue(v::Array)
```

Comment:
This can be written more simply as `PValue(v::AbstractVector) = convert(Vector{PValue}, v)`. But do you really need this function? Why not call `convert` directly?